crates/ruff_python_semantic/src/model.rs

```diff
@@ -1507,6 +1507,35 @@ impl<'a> SemanticModel<'a> {
             None
         })
     }
+
+    /// Similar to [`SemanticModel::lookup_attribute`], but does not resolve the scope.
+    pub fn resolve_load_attribute(&mut self, attr: &ast::Identifier, class_scope_id: ScopeId) {
+        let Some(class_scope) = self.scopes.get(class_scope_id) else {
+            return;
+        };
+        let Some(class_def) = class_scope.kind.as_class() else {
+            return;
+        };
+
+        // If class has base classes, the attribute might be defined in one of them.
+        // Since we don't want to recursively into the base classes, we skip resolving.
+        if !class_def.bases().is_empty() {
```

Comment:
I don't know the correct solution here but I think we can add class attributes form parent to children when visiting the class. But this would be out of scope of this PR.